InstanceManager.js

```diff
@@ -513,30 +606,31 @@ class InstanceManager {
     });
   }
 
-  shutdownCluster() {
+  async shutdownCluster() {
+    debugLog(`Shutting down the cluster`);
     const nonAgents = [...this.coordinators(), ...this.dbServers(), ...this.singleServers()];
 
-    return Promise.all(nonAgents.map(this.shutdown.bind(this)))
-    .then(() => {
-      return Promise.all(this.agents().map(this.shutdown.bind(this)))
-    });
+    let allWaiters = nonAgents.map(n => this.shutdown(n));
+    for (let waiter of allWaiters) {
```

Comment:
`allWaiters` can be inlined here. The target of `of` is only evaluated once, not on every loop. Same goes with the next one below.